internal/server/data/migrations.go

```diff
@@ -222,3 +226,73 @@ func dropDeletedProviderUsers() *migrator.Migration {
 		},
 	}
 }
+
+// addFieldsFor_0_14_3 adds all migrations that were previously applied by a
+// second call to gorm.AutoMigrate. In this release we're removing the
+// unconditional call to gorm.AutoMigrate in favor of having explicit migrations
+// for all changes.
+//
+// To account for all the existing migrations that were applied by AutoMigrate
+// we have to call it here again on any tables that have had changes.
+//
+// In the future we should use ALTER TABLE sql statements instead of AutoMigrate.
+//
+// nolint:revive
+func addFieldsFor_0_14_3() *migrator.Migration {
+	return &migrator.Migration{
+		ID: "2022-07-21T18:28",
+		Migrate: func(tx *gorm.DB) error {
+			if err := tx.AutoMigrate(&models.Provider{}); err != nil {
+				return err
+			}
+			if err := tx.AutoMigrate(&models.Settings{}); err != nil {
+				return err
+			}
+			if err := tx.AutoMigrate(&models.Organization{}); err != nil {
+				return err
+			}
+			if err := tx.AutoMigrate(&models.AccessKey{}); err != nil {
+				return err
+			}
+			if err := tx.AutoMigrate(&models.Credential{}); err != nil {
+				return err
+			}
+			if err := tx.AutoMigrate(&models.Destination{}); err != nil {
+				return err
+			}
+			if err := tx.AutoMigrate(&models.ProviderUser{}); err != nil {
+				return err
+			}
+			if err := tx.AutoMigrate(&models.Group{}); err != nil {
+				return err
+			}
+
+			if err := tx.Exec(`
+ALTER TABLE credentials DROP COLUMN IF EXISTS one_time_password_used;
+
+ALTER TABLE provider_users DROP COLUMN IF EXISTS id;
+ALTER TABLE provider_users DROP COLUMN IF EXISTS created_at;
+ALTER TABLE provider_users DROP COLUMN IF EXISTS updated_at;
+`).Error; err != nil {
+				return err
+			}
+
+			if !tx.Migrator().HasConstraint("provider_users", "provider_users_pkey") {
+				if err := tx.Exec(`
+ALTER TABLE ONLY provider_users
```

Comment:
I believe this doesn't work on sqlite as it doesn't allow many options for table alteration, if that's the case is that intentional?

Reply:
Ya, good call out. I believe I changed the migration test to only postgres because of this. I should check if we're ready to discontinue sqlite support for production use cases.